### ISSUE TITLE
checkbox - WCAG: (1.3.1, 3.3.2) - input must have labels

### ISSUE THREAD

**Issue by JoaoEnesGoncalves** [lumada-design/hv-uikit-react#718]
**Describe the bug**
In case of simple checkbox, example when integrated in table, violated the accessibility standard WCAG: 1.3.1, 3.3.2

**To Reproduce**
1. Run
`pa11y --runner htmlcs --runner axe --debug --reporter json --standard WCAG2AA --root-element "#root > div.Component-content-4" http://localhost:9002/iframe.html?id=corecheckbox--checkboxdisabled`

**Error**
```
  "code": "label",
  "type": "error",
  "typeCode": 1,
  "message": "Form elements must have labels (https://dequeuniversity.com/rules/axe/3.3/label?application=axeAPI)",
  "context": "<input class=\"MuiPrivateSwitchBase-input-170\" disabled=\"\" type=\"checkbox\" data-indeterminate=\"false\" value=\"\">",
  "selector": "#root > div:nth-child(2) > div > div > label > span:nth-child(1) > span > input",
  "runner": "axe",
  "runnerExtras": {
    "description": "Ensures every form element has a label",
    "impact": "critical",
    "help": "Form elements must have labels",
    "helpUrl": "https://dequeuniversity.com/rules/axe/3.3/label?application=axeAPI"
```
